File: app/src/main/java/com/tunjid/androidbootstrap/fragments/HidingViewFragment.java
package com.tunjid.androidbootstrap.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.tunjid.androidbootstrap.R;
import com.tunjid.androidbootstrap.baseclasses.AppBaseFragment;
import com.tunjid.androidbootstrap.core.view.ViewHider;

/**
 * Fragment showing a static list of images
 * <p>
 * Created by tj.dahunsi on 5/6/17.
 */

public class HidingViewFragment extends AppBaseFragment
        implements View.OnClickListener {

    private ViewHider leftHider;
    private ViewHider topHider;
    private ViewHider rightHider;
    private ViewHider bottomHider;

    public static HidingViewFragment newInstance() {
        HidingViewFragment fragment = new HidingViewFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_hiding_view, container, false);

        View leftButton = rootView.findViewById(R.id.left_button);
        View topButton = rootView.findViewById(R.id.top_button);
        View rightButton = rootView.findViewById(R.id.right_button);
        View bottomButton = rootView.findViewById(R.id.bottom_button);

        leftButton.setOnClickListener(this);
        topButton.setOnClickListener(this);
        rightButton.setOnClickListener(this);
        bottomButton.setOnClickListener(this);
        rootView.findViewById(R.id.reset).setOnClickListener(this);

        leftHider = new ViewHider(leftButton, ViewHider.LEFT);
        topHider = new ViewHider(topButton, ViewHider.TOP);
        rightHider = new ViewHider(rightButton, ViewHider.RIGHT);
        bottomHider = new ViewHider(bottomButton, ViewHider.BOTTOM);

        return rootView;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.left_button:
                leftHider.hide();
                break;
            case R.id.top_button:
                topHider.hide();
                break;
            case R.id.right_button:
                rightHider.hide();
                break;
            case R.id.bottom_button:
                bottomHider.hide();
                break;
            case R.id.reset:
                leftHider.show();
                topHider.show();
                rightHider.show();
                bottomHider.show();
                break;
        }
    }
}

File: app/src/main/java/com/tunjid/androidbootstrap/fragments/HidingViewFragment.kt
package com.tunjid.androidbootstrap.fragments

import android.os.Bundle
import android.view.View
import androidx.core.content.ContextCompat
import com.tunjid.androidbootstrap.uidrivers.GlobalUiController
import com.tunjid.androidbootstrap.R
import com.tunjid.androidbootstrap.uidrivers.UiState
import com.tunjid.androidbootstrap.uidrivers.activityGlobalUiController
import com.tunjid.androidbootstrap.baseclasses.AppBaseFragment
import com.tunjid.androidbootstrap.view.animator.ViewHider

/**
 * Fragment demonstrating hiding views
 *
 *
 * Created by tj.dahunsi on 5/6/17.
 */

class HidingViewFragment : AppBaseFragment(R.layout.fragment_hiding_view), GlobalUiController {

    override var uiState: UiState by activityGlobalUiController()

    private lateinit var leftHider: ViewHider<View>
    private lateinit var topHider: ViewHider<View>
    private lateinit var rightHider: ViewHider<View>
    private lateinit var bottomHider: ViewHider<View>

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        uiState = uiState.copy(
                toolbarTitle = this::class.java.simpleName,
                toolbarShows = true,
                toolBarMenu = 0,
                fabShows = false,
                showsBottomNav = false,
                navBarColor = ContextCompat.getColor(requireContext(), R.color.white_75)
        )

        val leftButton = view.findViewById<View>(R.id.left_button)
        val topButton = view.findViewById<View>(R.id.top_button)
        val rightButton = view.findViewById<View>(R.id.right_button)
        val bottomButton = view.findViewById<View>(R.id.bottom_button)

        leftButton.setOnClickListener(this::onClick)
        topButton.setOnClickListener(this::onClick)
        rightButton.setOnClickListener(this::onClick)
        bottomButton.setOnClickListener(this::onClick)
        view.findViewById<View>(R.id.reset).setOnClickListener(this::onClick)

        leftHider = ViewHider.of(leftButton).setDirection(ViewHider.LEFT).build()
        topHider = ViewHider.of(topButton).setDirection(ViewHider.TOP).build()
        rightHider = ViewHider.of(rightButton).setDirection(ViewHider.RIGHT).build()
        bottomHider = ViewHider.of(bottomButton).setDirection(ViewHider.BOTTOM).build()
    }

    private fun onClick(v: View) {
        when (v.id) {
            R.id.left_button -> leftHider.hide()
            R.id.top_button -> topHider.hide()
            R.id.right_button -> rightHider.hide()
            R.id.bottom_button -> bottomHider.hide()
            R.id.reset -> {
                leftHider.show()
                topHider.show()
                rightHider.show()
                bottomHider.show()
            }
        }
    }

    companion object {
        fun newInstance(): HidingViewFragment = HidingViewFragment().apply { arguments = Bundle() }
    }
}
